File: tempconv/tempconv.java
package tempconv;
public class tempconv {
    public static class Celsius {
        private double value;

        public Celsius(double value) {
            this.value = value;
        }

        public double getValue() {
            return value;
        }

        @Override
        public String toString() {
            return value + "°C";
        }
    }

    public static class Fahrenheit {
        private double value;

        public Fahrenheit(double value) {
            this.value = value;
        }

        public double getValue() {
            return value;
        }

        @Override
        public String toString() {
            return value + "°F";
        }
    }

    public static final Celsius AbsoluteZeroC = new Celsius(-273.15);
    public static final Celsius FreezingC = new Celsius(0);
    public static final Celsius BoilingC = new Celsius(100);

    public static Fahrenheit celsiusToFahrenheit(Celsius celsius) {
        double value = celsius.getValue() * 9/5 + 32;
        return new Fahrenheit(value);
    }

    public static Celsius fahrenheitToCelsius(Fahrenheit fahrenheit) {
        double value = (fahrenheit.getValue() - 32) * 5/9;
        return new Celsius(value);
    }

  }

File: tempconv/tempconv.cs
# include <stdio.h>

typedef double Celsius;
typedef double Fahrenheit;

const Celsius AbsoluteZeroC = -273.15;
const Celsius FreezingC = 0;
const Celsius BoilingC = 100;

void printCelsius(Celsius c)
{
    printf("%.2f°C\n", c);
}

void printFahrenheit(Fahrenheit f)
{
    printf("%.2f°F\n", f);
}

Fahrenheit celsiusToFahrenheit(Celsius c)
{
    return c * 9 / 5 + 32;
}

Celsius fahrenheitToCelsius(Fahrenheit f)
{
    return (f - 32) * 5 / 9;
}

int main()
{
    Celsius c = -10.0;
    Fahrenheit f = 32.0;

    printf("Celsius: ");
    printCelsius(c);

    printf("Fahrenheit: ");
    printFahrenheit(f);

    printf("Celsius to Fahrenheit: ");
    printFahrenheit(celsiusToFahrenheit(c));

    printf("Fahrenheit to Celsius: ");
    printCelsius(fahrenheitToCelsius(f));

    return 0;
}
